### ISSUE TITLE
Migrate Giant Swarm workload clusters to Cluster API for AWS

### ISSUE THREAD

**Issue by alex-dabija** [giantswarm/roadmap#309]
## User Story

_- As a user, I want to reconcile existing Giant Swarm clusters with Cluster API for AWS controllers clusters instead of the Giant Swarm controllers so that I can leverage the flexibility and extra features available in Cluster API._

## Details, Background

It's very important to have a migration path for the existing Giant Swarm clusters to Cluster API in order not to force a disruptive move for our customers.

The migration risk can be minimized by identifying the differences between existing Giant Swarm clusters and Cluster API clusters as soon as possible.

## Blocked by / depends on

None

**Comment by alex-dabija** [comment]
Duplicate of https://github.com/giantswarm/roadmap/issues/1787 .
